replica/table.cc

```diff
@@ -2937,13 +2937,31 @@ table::query(schema_ptr s,
         co_return make_lw_shared<query::result>();
     }
 
-    _async_gate.enter();
+    // 1 for the table and >= 1 for the compaction groups that overlap
+    utils::small_vector<gate::holder, 2> async_gate_holders;
+    async_gate_holders.push_back(_async_gate.hold());
+    {
```

Comment:
why the nested block?

Reply:
To free `token_range`, it is not required after checking the overlaps.